lib/transaction.js

```diff
@@ -85,7 +85,7 @@ Transaction.begin = function(connector, options, cb) {
       return cb(err);
     }
     var tx = connection;
-    if (!(connection instanceof Transaction)) {
+    if (!(connection instanceof Transaction) && !(typeof connection.txId === 'string')) {
```

Comment:
ok I think you are suggesting the same thing(duck-typing check) as Raymond said in https://github.com/strongloop/loopback-connector-postgresql/pull/268#issuecomment-311809679

I misunderstood his idea, I thought he suggests check both(instance and txId), while he is probably actually suggesting check some property that **shared** across all connectors, and could also **prove** the `connector` is a `Transaction` instance